node/node.go

```diff
@@ -501,22 +502,42 @@ func (n *Node) Dispatch() error {
 
 func (n *Node) initDatabase() error {
 	// start the db
-	switch n.Config.DatabaseConfig.Name {
-	case leveldb.Name:
-		dbPath := filepath.Join(n.Config.DatabaseConfig.Path, version.CurrentDatabase.String())
-		var err error
-		n.DB, err = leveldb.New(dbPath, n.Config.DatabaseConfig.Config, n.Log, "db_internal", n.MetricsRegisterer)
+	switch dbName := n.Config.DatabaseConfig.Name; dbName {
+	case memdb.Name:
+		n.DB = memdb.New()
+	case leveldb.Name, pebble.Name:
+		var (
+			finalPathElem = pebble.Name
+			newDBFunc     = pebble.New
+		)
+		if dbName == leveldb.Name {
+			// Prior to v1.10.15, the only on-disk database was leveldb,
+			// and its files went to [dbPath]/[networkID]/v1.4.5.
+			finalPathElem = version.CurrentDatabase.String()
+			newDBFunc = leveldb.New
+		}
+
+		var (
+			dbPath = filepath.Join(n.Config.DatabaseConfig.Path, finalPathElem)
+			err    error
+		)
+		n.DB, err = newDBFunc(
+			dbPath,
+			n.Config.DatabaseConfig.Config,
+			n.Log,
+			"db_internal",
+			n.MetricsRegisterer,
+		)
 		if err != nil {
 			return fmt.Errorf("couldn't create db at %s: %w", dbPath, err)
 		}
```

Comment:
I feel like this is more clear if we just explicitly handle the different DBs.
```suggestion
	switch n.Config.DatabaseConfig.Name {
	case leveldb.Name:
		dbPath := filepath.Join(n.Config.DatabaseConfig.Path, version.CurrentDatabase.String())
		var err error
		n.DB, err = leveldb.New(dbPath, n.Config.DatabaseConfig.Config, n.Log, "db_internal", n.MetricsRegisterer)
		if err != nil {
			return fmt.Errorf("couldn't create leveldb at %s: %w", dbPath, err)
		}
	case memdb.Name:
		n.DB = memdb.New()
	case pebble.Name:
		dbPath := filepath.Join(n.Config.DatabaseConfig.Path, pebble.Name)
		var err error
		n.DB, err = pebble.New(dbPath, n.Config.DatabaseConfig.Config, n.Log, "db_internal", n.MetricsRegisterer)
		if err != nil {
			return fmt.Errorf("couldn't create pebbledb at %s: %w", dbPath, err)
		}
```

Reply:
We could keep the:
```
// Prior to v1.10.15, the only on-disk database was leveldb,
// and its files went to [dbPath]/[networkID]/v1.4.5.
```
on level tho... that's probably nice